plugins/developer_guide.md

```diff
@@ -0,0 +1,305 @@
+All extensions registered using this entry point will be invoked.
+Each of them is required to implement `DocumentableTransformer` interface:
+
+```kotlin
+interface DocumentableTransformer {
+    operator fun invoke(original: DModule, context: DokkaContext): DModule
+}
+```
+
+By default, `InheritorsExtractorTransformer` is created, that extracts inherited classes data across source sets and creates inheritance map.
+
+#### Creating page models
+
+The documentable model is translated into page format, that aggregates all tha data that will be available for different pages of documentation.
+
+This step uses `DokkaCore.documentableToPageTranslator` entry point.
+It is required to have exactly one extension registered for this entry point.
+Having more will trigger an error, unless all except one will have parameter `isFallback` set to true, in which case the non-fallback extension will be used.
+
+The extension is required to implement `DocumentableToPageTranslator` interface:
+
+```kotlin
+interface DocumentableToPageTranslator {
+    operator fun invoke(module: DModule): ModulePageNode
+}
+```
+
+By default, `DefaultDocumentableToPageTranslator` is created.  This extension is a fallback, so it could be replaced by a custom non-fallback one.
+
+#### Transforming page models
+
+You can apply any transformations to paged data.
+
+This step uses `DokkaCore.pageTransformer` entry point.
+All extensions registered using this entry point will be invoked.
+Each of them is required to implement `PageTransformer` interface:
+
+```kotlin
+interface PageTransformer {
+    operator fun invoke(input: RootPageNode): RootPageNode
+}
+```
+By default, two transformers are created:
+* `PageMerger` merges some pages depending on `MergeStrategy`
+* `DeprecatedStrikethroughTransformer` marks all deprecated members on every page
+
+#### Rendering
+
+All pages are rendered to desired format.
+
+This step uses `DokkaCore.renderer` entry point.
+It is required to have exactly one extension registered for this entry point.
+Having more will trigger an error, unless all except one will have parameter `isFallback` set to true, in which case the non-fallback extension will be used.
+                                    
+The extension is required to implement  `Renderer` interface:
+
+```kotlin
+interface Renderer {
+    fun render(root: RootPageNode)
+}
+```
+
+By default, only `HTMLRenderer`, that extends basic `DefaultRenderer`, is created, but it will be registered only if configuration parameter `format` is set to `html`.
+
+### Default extensions' extension points
+
+Default core extension points already have an implementation for providing basic dokka functionality. All of them are declared in `DokkaBase` plugin.
+If you don't want this default extensions to load, all you need to do is write a custom configuration to disable loading said plugin. You will then need to implement extensions for all core extension points. 
```

Comment:
What if I want only some to load? Or add a new extension to a current list of extensions for a extension point?

Also: how to disable them?

Reply:
I added a paragraph about adding extension points and obtaining extension instances.

You can not load only some of them.